base64ct/src/lib.rs

```diff
@@ -311,33 +529,27 @@ fn encode_3bytes_padded(src: &[u8], dst: &mut [u8]) {
 }
 
 #[inline(always)]
-fn encode_6bits(src: i16) -> u8 {
+fn encode_6bits(src: i16, hi_bytes: (u8, u8)) -> u8 {
+    let hi_off = 0x1c + (hi_bytes.0 & 4);
```

Comment:
I ended up addressing the off-by-4 issue by introducing this tweak.

It works for the Base64/Base64url alphabets but isn't generalizable. I'm also not entirely confident in it, so it'd be good to add some better tests (possibly `proptest` against the `base64` crate)

Reply:
Would be curious to know if @Sc00bz has a better solution for this problem